samza-kafka/src/test/scala/org/apache/samza/config/TestKafkaConfig.scala

```diff
@@ -195,25 +195,136 @@ class TestKafkaConfig {
 
   @Test
   def testChangeLogReplicationFactor() {
+    props.setProperty("stores.store-with-override.changelog", "kafka-system.changelog-topic")
     props.setProperty("stores.store-with-override.changelog.replication.factor", "3")
 
     val mapConfig = new MapConfig(props.asScala.asJava)
     val kafkaConfig = new KafkaConfig(mapConfig)
-    assertEquals(kafkaConfig.getChangelogStreamReplicationFactor("store-with-override"), "3")
-    assertEquals(kafkaConfig.getChangelogStreamReplicationFactor("store-without-override"), "2")
-    assertEquals(kafkaConfig.getDefaultChangelogStreamReplicationFactor , "2")
+    assertEquals("3", kafkaConfig.getChangelogStreamReplicationFactor("store-with-override"))
+    assertEquals("2", kafkaConfig.getChangelogStreamReplicationFactor("store-without-override"))
+    assertEquals("2", kafkaConfig.getDefaultChangelogStreamReplicationFactor)
   }
 
   @Test
   def testChangeLogReplicationFactorWithOverriddenDefault() {
+    props.setProperty(JobConfig.JOB_DEFAULT_SYSTEM, "kafka-system")
+    props.setProperty("stores.store-with-override.changelog", "changelog-topic")
     props.setProperty("stores.store-with-override.changelog.replication.factor", "4")
     // Override the "default" default value
     props.setProperty("stores.default.changelog.replication.factor", "5")
 
     val mapConfig = new MapConfig(props.asScala.asJava)
     val kafkaConfig = new KafkaConfig(mapConfig)
-    assertEquals(kafkaConfig.getChangelogStreamReplicationFactor("store-with-override"), "4")
-    assertEquals(kafkaConfig.getChangelogStreamReplicationFactor("store-without-override"), "5")
-    assertEquals(kafkaConfig.getDefaultChangelogStreamReplicationFactor , "5")
+    assertEquals("4", kafkaConfig.getChangelogStreamReplicationFactor("store-with-override"))
+    assertEquals("5", kafkaConfig.getChangelogStreamReplicationFactor("store-without-override"))
+    assertEquals("5", kafkaConfig.getDefaultChangelogStreamReplicationFactor)
+  }
+
+  @Test
+  def testChangeLogReplicationFactorWithSystemOverriddenDefault() {
+    props.setProperty(StorageConfig.CHANGELOG_SYSTEM, "kafka-system")
+    props.setProperty("systems.kafka-system.default.stream.replication.factor", "8")
+    props.setProperty("stores.store-with-override.changelog.replication.factor", "4")
+
+    val mapConfig = new MapConfig(props.asScala.asJava)
+    val kafkaConfig = new KafkaConfig(mapConfig)
+    assertEquals("4", kafkaConfig.getChangelogStreamReplicationFactor("store-with-override"))
+    assertEquals("8", kafkaConfig.getChangelogStreamReplicationFactor("store-without-override"))
+    assertEquals("8", kafkaConfig.getDefaultChangelogStreamReplicationFactor)
+  }
+
+  @Test
+  def testCheckpointReplicationFactor() {
+    val emptyConfig = new KafkaConfig(new MapConfig())
+    assertNull(emptyConfig.getCheckpointReplicationFactor.orNull)
```

Comment:
Seems like a misunderstanding. There is no default for checkpoint. 

Defaults in terms of the config class:
coordinator - 3
changelog - 2
checkpoint - null

"But", I hear you say, "the config table clearly says that the default checkpoint replication factor is 3". Yeah, *sigh*, it's another example of how muddled the configs have gotten. Apparently the default for checkpoints is injected in KafkaCheckpointManagerFactory, whereas the defaults for coordinator/changelog are defaulted in the config classes (and sometimes in down stream classes... for good measure? IDK).

Anyway, I agree with the consistency argument. I missed that while doing this change. Will fix it. Thanks!